src/simulation_manager/mod.rs

```diff
@@ -0,0 +1,49 @@
+use village_manager::VillageManager;
+use village_manager::village::*;
+use village_manager::village_mind::*;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+pub struct SimulationManager {
+    village_managers: Vec<VillageManager>,
+}
+
+impl SimulationManager {
+    pub fn new(number_of_villages: u32) -> SimulationManager {
```

Comment:
Instead of specifying number of villages, move to a create_village(Village) model instead.